curriculum/challenges/english/25-front-end-development/lab-contact-form/66f26c32ec6f90df01a44f60.md

```diff
@@ -0,0 +1,458 @@
+
+const labels = document.querySelectorAll('.form-container form label');
+
+assert.isAbove(labels.length, 0);
+
+let found = false;
+
+for (let label of labels) {
+    if (label.getAttribute('for') === emailInput) {
+        found = true;
+        break;
+    }
+}
+
+assert(found);
+```
+
+You should have a `label` element for the `#message` `textarea` with a `for` attribute matching the `id` value on the `textarea`.
+
+```js
+const messageInput = document.querySelector('.form-container form textarea')?.getAttribute('id');
+
+const labels = document.querySelectorAll('.form-container form label');
+
+assert.isAbove(labels.length, 0);
+
+let found = false;
+
+for (let label of labels) {
+    if (label.getAttribute('for') === messageInput) {
+        found = true;
+        break;
+    }
+}
+
+assert(found);
+```
+
+The labels shouldn't be empty.
+
+```js
+const labels = document.querySelectorAll('.form-container form label');
+
+assert.isAbove(labels.length, 0);
+
+for (let label of labels) {
+    const labelText = label.textContent.trim();
+    assert.isAbove(labelText.length, 0);
+}
+
+```
+
+The `form-container` `div` should have a background color.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.form-container')?.backgroundColor);
+```
+
+The `form-container` `div` should have a `border-radius`.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.form-container')?.borderRadius);
+```
+
+The `form-container` `div` should have padding.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.form-container')?.padding);
+```
+
+The `form-container` `div` should have a width.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.form-container')?.width);
+```
+
+Your labels should have a margin and font color.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('label')?.margin);
+assert(new __helpers.CSSHelp(document).getStyle('label')?.color);
+```
+
+Your `input` elements should have a width, padding, and a margin on the bottom.
+
+```js
+// Helper function to extract properties from multiple selectors
+function getStyleProperties(selectors, properties) {
+  const result = {};
+  properties.forEach((property) => {
+    result[property] = selectors.map((selector) => selector?.[property]).find(Boolean);
+  });
+  return result;
+}
+
+// Initializing selectors
+const selectors = [
+  new __helpers.CSSHelp(document).getStyle('input'),
+  new __helpers.CSSHelp(document).getStyle('input, textarea'),
+  new __helpers.CSSHelp(document).getStyle('textarea, input')
+];
+
+// Extract width, padding, and margin properties
+const { width } = getStyleProperties(selectors, ['width']);
+assert(width);
+
+const {
+  padding,
+  paddingTop,
+  paddingBottom,
+  paddingLeft,
+  paddingRight
+} = getStyleProperties(selectors, ['padding', 'paddingTop', 'paddingBottom', 'paddingLeft', 'paddingRight']);
+
+// Assert padding properties
+assert(
+  padding ||
+  (paddingTop && paddingBottom && paddingLeft && paddingRight)
+);
```

Comment:
```suggestion
assert.isOk(
  padding ||
  (paddingTop && paddingBottom && paddingLeft && paddingRight)
);
```

for this one, I have no idea for a more specific method. `isOk` checks that the value is truthy, so it seems the more appropriate

Reply:
Updated this, it passed.